spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -610,10 +621,12 @@ public void afterReceiveCompletion(Message<?> message, MessageChannel channel, E
 		}
 
 		public boolean remove(ChannelInterceptor interceptor) {
+			this.size--;
 			return this.interceptors.remove(interceptor);
```

Comment:
This is a bug!
The requested `interceptor` might not be in the list, so we should decrement only if `true` from `remove()`.